### PR TITLE
Convert to regression

### PR DESCRIPTION
**Pull request by kriscon-db**
## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

(Please fill in changes proposed in this fix)

## How is this patch tested?

- [ ] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [x] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/pipelines/regression/v1/pipeline.py [hunk 1]
```diff
@@ -159,7 +171,19 @@ class RegressionPipeline(_BasePipeline):
...
+        PredictStep,
+        # Training data ingestion DAG
+        IngestStep,
+        # Model training DAG
+        SplitStep,
+        TransformStep,
+        TrainStep,
+        EvaluateStep,
+        RegisterStep,
+    )
```
Comment (jerrylian-db) [inline mlflow/pipelines/regression/v1/pipeline.py:186]: With this setup, does this mean that when the entire MLP regression pipeline is run, all of the steps, including the Batch Scoring DAG will be executed?

File: mlflow/pipelines/regression/v1/pipeline.py [hunk 2]
```diff
@@ -523,6 +578,19 @@ def read_dataframe(artifact_name, output_dir, file_name, step_name):
                 log_artifact_not_found_warning("registered_model_version", register_step.name)
                 return None
 
+        elif artifact_name == "batch_scoring_data":
+            return read_dataframe(
+                "batch_scoring_data",
+                ingest_scoring_output_dir,
+                IngestScoringStep._DATASET_OUTPUT_NAME,
+                ingest_scoring_step.name,
+            )
```
Comment (jerrylian-db) [inline mlflow/pipelines/regression/v1/pipeline.py:587]: `"batch_scoring_data"` reads to me like the output the entire Batch Scoring DAG. Perhaps we can rename this? `"ingested_scoring_data"` is an idea

File: mlflow/pipelines/steps/ingest/__init__.py [hunk 3]
```diff
@@ -170,6 +170,52 @@ def from_pipeline_config(cls, pipeline_config: Dict[str, Any], pipeline_root: st
             pipeline_root=pipeline_root,
         )
 
+
+class IngestStep(BaseIngestStep):
```
Comment (jerrylian-db) [inline mlflow/pipelines/steps/ingest/__init__.py:174]: Smart approach! I like it :)

File: mlflow/pipelines/steps/predict.py [hunk 4]
```diff
@@ -30,7 +30,7 @@ def _build_profiles_and_card(self, scored_df) -> BaseCard:
         # Build profiles for input dataset, and train / validation / test splits
         scored_data_profile = get_pandas_data_profile(
             scored_df.reset_index(drop=True),
-            "Profile of Preprocessed Dataset",
+            "Profile of Scored Dataset",
```
Comment (jerrylian-db) [inline mlflow/pipelines/steps/predict.py:33]: Thank you for catching this!

**Review by jerrylian-db — COMMENTED** [review]
Thanks so much for your hard work to adapt our batch scoring work to the pivot! Just some minor comments.

File: mlflow/pipelines/regression/v1/pipeline.py (hunk 1 above)
Comment (kriscon-db) [reply mlflow/pipelines/regression/v1/pipeline.py:186]: No.  I asked about this in a thread in channel.  For now, we can leave this as is and Batch Scoring will need to be explicitly called.

File: mlflow/pipelines/regression/v1/pipeline.py (hunk 2 above)
Comment (kriscon-db) [reply mlflow/pipelines/regression/v1/pipeline.py:587]: Agree.  Let me fix this.

**Review by jerrylian-db — APPROVED** [review]
LGTM. I think we can merge this after fixing tests.

**Pull request merged**